models/packages/package_version.go

```diff
@@ -185,19 +150,24 @@ func HasVersionFileReferences(ctx context.Context, versionID int64) (bool, error
 
 // PackageSearchOptions are options for SearchXXX methods
 type PackageSearchOptions struct {
-	OwnerID      int64
-	RepoID       int64
-	Type         string
-	PackageID    int64
-	QueryName    string
-	QueryVersion string
-	Properties   map[string]string
-	Sort         string
+	OwnerID           int64
+	RepoID            int64
+	Type              Type
+	PackageID         int64
+	NameExactMatch    bool
+	QueryName         string
+	VersionExactMatch bool
+	QueryVersion      string
+	Properties        map[string]string
+	IsInternal        bool
+	HasFileWithName   string
+	HasFiles          util.OptionalBool
+	Sort              string
```

Comment:
I think these options need some comments explaining what they do. Certainly the QueryName and QueryVersion need a comment to explain that without _ExactMatch set these are LIKE strings.

I guess I also question as to why QueryName and QueryVersion get the prefix Query whereas the rest don't.